src/common/file_utils.cpp

```diff
@@ -72,6 +77,64 @@ std::unique_ptr<FileInfo> FileUtils::openFile(const std::string& path, int flags
 #endif
 }
 
+void FileUtils::openFile(const std::string& path, uint8_t flags, FileLockType lock_type) {
+    int open_flags = 0;
+	int rc;
+	bool open_read = flags & FileFlags::FILE_FLAGS_READ;
+	bool open_write = flags & FileFlags::FILE_FLAGS_WRITE;
+	if (open_read && open_write) {
+		open_flags = O_RDWR;
+	} else if (open_read) {
+		open_flags = O_RDONLY;
+	} else if (open_write) {
+		open_flags = O_WRONLY;
+	} else {
+        throw Exception("READ, WRITE or both should be specified when opening a file");	
+    }
+	if (open_write) {
+		// need Read or Write
+		assert(flags & FileFlags::FILE_FLAGS_WRITE);
+		open_flags |= O_CLOEXEC;
+		if (flags & FileFlags::FILE_FLAGS_FILE_CREATE) {
+			open_flags |= O_CREAT;
+		} else if (flags & FileFlags::FILE_FLAGS_FILE_CREATE_NEW) {
+			open_flags |= O_CREAT | O_TRUNC;
+		}
+		if (flags & FileFlags::FILE_FLAGS_APPEND) {
+			open_flags |= O_APPEND;
+		}
+	}
+	if (flags & FileFlags::FILE_FLAGS_DIRECT_IO) {
+#if defined(__sun) && defined(__SVR4)
+		throw Exception("DIRECT_IO not supported on Solaris");
+#endif
+#if defined(__DARWIN__) || defined(__APPLE__) || defined(__OpenBSD__)
+		// OSX does not have O_DIRECT, instead we need to use fcntl afterwards to support direct IO
+		open_flags |= O_SYNC;
+#else
+		open_flags |= O_DIRECT | O_SYNC;
+#endif
+	}
+    int fd = open (path.c_str(), open_flags, 0666);//(path.c_str(), O_RDWR);
+    // (path.c_str(), flags, 0666);
+    if (fd == -1) {
+        throw Exception("Cannot open file: " + path);
+    }
+
+    if (lock_type != FileLockType::NO_LOCK) {
```

Comment:
Let's not add a new `openFile` function to `FileUtils`. I believe we can merge the lock logic into existing `openFile` function. By default, `openFile` will opens a file in `NO_LOCK` mode.

Also, remember we should handle windows too (if it is doable).